core/src/main/java/tech/tablesaw/columns/Column.java

```diff
@@ -564,6 +564,15 @@ default boolean noneMatch(Predicate<? super T> test) {
 
   Column<T> set(int row, T value);
 
+  default Column<T> set(int row, String newValue, AbstractColumnParser<?> parser) {
+    AbstractColumnParser<T> typedParser = (AbstractColumnParser<T>) parser;
+    if (parser.isMissing(newValue)) {
+      return setMissing(row);
+    } else {
+      return set(row, typedParser.parse(newValue));
```

Comment:
this is still going to do autoboxing because it calls the generic `parse` instead of `parseDouble`, `parseFloat`, etc.

Reply:
Implemented primitive type parsing for Column types that need it.